File: src/Aoe2BuildOrders.Api/Models/BuildOrder.cs
namespace Aoe2BuildOrders.Api.Models;

public class BuildOrder
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public required string Civilization { get; set; }
    public required string StrategyType { get; set; }
    public required string Difficulty { get; set; }
    public string? Description { get; set; }

    public List<BuildOrderStep> Steps { get; set; } = [];
}

File: client/src/types/buildOrder.ts
export type BuildOrderSummary = {
  id: number;
  name: string;
  civilization: string;
  strategyType: string;
  difficulty: string;
  description?: string;
  stepCount: number;
};

export type BuildOrderStep = {
  id: number;
  buildOrderId: number;
  stepNumber: number;
  population?: number;
  age: string;
  instruction: string;
  resourceFocus?: string;
  notes?: string;
};

export type BuildOrderDetail = {
  id: number;
  name: string;
  civilization: string;
  strategyType: string;
  difficulty: string;
  description?: string;
  steps: BuildOrderStep[];
};
